### ISSUE TITLE
Replace old color system variables

### ISSUE THREAD

**Issue by bonniezhou** [material-components/material-components-web#3124]
Follow up to #2762.

Material Design has an updated [color system](https://material.io/design/color/the-color-system.html#color-theme-creation). It replaces the `text-<TEXT_STYLE>-on-<BACKGROUND>` variables with the color variables `primary`, `secondary`, `background`, `surface` and `error`, plus the variable `on-*` for text on each of those 5 colors.

- [ ] remove all instances of `text-<TEXT_STYLE>-on-<BACKGROUND>` variables
- [ ] add `primary`, `secondary`, `background`, `surface` and `error`, plus `on-*` for text
- [ ] add `high-emphasis`, `medium-emphasis`, `disabled-ink` and `disabled-fill` variables for opacity values (variable names pending design approval) 

Here's a [sample branch](https://github.com/material-components/material-components-web/compare/refactor/theme/color) for the refactor.

**Comment by kfranqueiro** [comment]
I expected the text-primary/etc. levels would be somewhat replaced by variables for different emphasis levels? Is that still supposed to happen?

**Comment by bonniezhou** [comment]
Yep, opacity values will also be encoded into variables. Updated description.

**Comment by kintz09** [comment]
Is there a recommended solution to set the emphasis level for text on [surface, primary, secondary, error] elements until this issue is completed?
